src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts

```diff
@@ -569,6 +569,7 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
         const jsonResponse: ProgrammingSubmission[] = res.body!;
         const body: ProgrammingSubmission[] = [];
         for (let i = 0; i < jsonResponse.length; i++) {
```

Comment:
can we use a modern for each loop here and properly name the elements?

Reply:
something like the following
```suggestion
        for (const submission of submissions ) {
```